package.json

```diff
@@ -3,7 +3,7 @@
   "displayName": "Apache Daffodil™ Extension for Visual Studio Code",
   "description": "Apache Daffodil™ Extension for Visual Studio Code providing DFDL syntax highlighting, DFDL code completion, DFDL schema debugging, and data editor",
   "version": "1.3.0",
-  "daffodilVersion": "3.4.0",
+  "daffodilVersion": "3.5.0",
```

Comment:
Daffodil had some license updates for 3.5.0. I think some or all of these need to get pulled into the debugger license files? Here's a command to show the changes to the Daffodil license files:

```
git diff v3.4.0 v3.5.0 -- LICENSE NOTICE daffodil-cli/bin.LICENSE daffodil-cli/bin.NOTICE 
```

Reply:
Would the changes under `daffodil-cli` be relevant to `daffodil-vscode`? We don't depend on it.

Otherwise the only thing that shows a difference is the copyright date.

Reply:
Some of the changes may be relevant. The Daffodil CLI license files are a good reference because they include the license information of all libraries that Daffodil depends on (since the CLI bundles all those libraries). So if you depend on Daffodil, it's likely you also depend on and include those same libraries and need to copy the license information.

Some exceptions are for libraries that are specific to the CLI like jline, jansi, and scallop that you likely don't depend on and don't need to include. And if you exclude libraries Daffodil depends on then you also don't need those.

But for example, the CLI pulls in jackson core for json support. The jackson license/notice changed, so unless the vscode debugger excludes that or doesn't pull it in, then you need to include those updates.

Reply:
We also removed log4j dependencies, I'm not sure what logging vscode uses but you can remove unless you are explicitly adding them.

We also added a scala-logging dependency.

Reply:
Hmm, those didn't appear in the diff command, so I missed them, so I'll poke around a bit more.

Reply:
Oh there's a `build/package` directory with its own `LICENSE` and `NOTICE` file... hmm.

Reply:
Did you include `daffodil-cli/bin.NOTICE` in the diff? THis is the diff I see for that file:

```diff
index 755b6a29d..dab451ff9 100644
--- a/daffodil-cli/bin.NOTICE
+++ b/daffodil-cli/bin.NOTICE
@@ -19,6 +19,10 @@ The following NOTICE information applies to binary components distributed with t
   been in development since 2007.
   It is currently developed by a community of developers.
 
+  ## Copyright
+
+  Copyright 2007-, Tatu Saloranta (tatu.saloranta@iki.fi)
+
   ## Licensing
 
   Jackson 2.x core and extension components are licensed under Apache License 2.0
@@ -30,6 +34,20 @@ The following NOTICE information applies to binary components distributed with t
   in some artifacts (usually source distributions); but is always available
   from the source code management (SCM) system project uses.
 
+  ## FastDoubleParser
+
+  jackson-core bundles a shaded copy of FastDoubleParser <https://github.com/wrandelshofer/FastDoubleParser>.
+  That code is available under an MIT license <https://github.com/wrandelshofer/FastDoubleParser/blob/main/LICENSE>
+  under the following copyright.
+
+  Copyright © 2023 Werner Randelshofer, Switzerland. MIT License.
+
+  See FastDoubleParser-NOTICE for details of other source code included in FastDoubleParser
+  and the licenses and copyrights that apply to that code.
+
+- com.typesafe.scala-logging.scala-logging_<VERSION>.jar
+  Copyright 2014-2021 Lightbend, Inc.
+
 - commons-codec.commons-codec-<VERSION>.jar
   Apache Commons Codec
   Copyright 2002-2020 The Apache Software Foundation
@@ -87,25 +105,6 @@ The following NOTICE information applies to binary components distributed with t
   Apache HttpComponents Core HTTP/2
   Copyright 2005-2020 The Apache Software Foundation
 
-- org.apache.logging.log4j.log4j-api-<VERSION>.jar
-  Apache Log4j API
-  Copyright 1999-2022 The Apache Software Foundation
-
-- org.apache.logging.log4j.log4j-api-scala_<VERSION>.jar
-  Apache Log4j Scala API
-  Copyright 2016-2022 Apache Software Foundation
-
-  AsciidocPlugin.scala
-  Copyright (c) 2008, 2009, 2010, 2011 Josh Suereth, Steven Blundy, Josh Cough,
-  Mark Harrah, Stuart Roebuck, Tony Sloane, Vesa Vilhonen, Jason Zaugg
-
-- org.apache.logging.log4j.log4j-core-<VERSION>.jar
-  Apache Log4j Core
-  Copyright 1999-2012 Apache Software Foundation
-
-  ResolverUtil.java
-  Copyright 2005-2006 Tim Fennell
-
 - org.jdom.jdom2-<VERSION>.jar
   Copyright (C) 2000-2012 Jason Hunter & Brett McLaughlin.
   All rights reserved.
@@ -134,8 +133,8 @@ The following NOTICE information applies to binary components distributed with t
 - org.scala-lang.scala-library-<VERSION>.jar
 - org.scala-lang.scala-reflect-<VERSION>.jar
   Scala
-  Copyright (c) 2002-2022 EPFL
-  Copyright (c) 2011-2022 Lightbend, Inc.
+  Copyright (c) 2002-2023 EPFL
+  Copyright (c) 2011-2023 Lightbend, Inc.
 
   Scala includes software developed at
   LAMP/EPFL (https://lamp.epfl.ch/) and
```

And here's the diff for `daffodil-cli/bin.LICENSE`
```diff
diff --git a/daffodil-cli/bin.LICENSE b/daffodil-cli/bin.LICENSE
index e9ac9701a..e0bae9ece 100644
--- a/daffodil-cli/bin.LICENSE
+++ b/daffodil-cli/bin.LICENSE
@@ -228,7 +228,7 @@ is subject to the terms and conditions of the following licenses.
 
     COPYRIGHT AND PERMISSION NOTICE
 
-    Copyright © 1991-2022 Unicode, Inc. All rights reserved.
+    Copyright © 1991-2023 Unicode, Inc. All rights reserved.
     Distributed under the Terms of Use in https://www.unicode.org/copyright.html.
 
     Permission is hereby granted, free of charge, to any person obtaining
@@ -651,6 +651,32 @@ is subject to the terms and conditions of the following licenses.
     (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
     OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+- com.fasterxml.jackson.core.jackson-core-2.13.4.jar
+  This product bundles 'Jackson JSON processor' which bundles a shaded copy of FastDoubleParser <https://github.com/wrandelshofer/FastDoubleParser>.
+  That code is available under the MIT license:
+
+    MIT License
+
+    Copyright (c) 2023 Werner Randelshofer, Switzerland.
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+
 - com.lihaoyi.geny_<VERSION>.jar
 - com.lihaoyi.os-lib_<VERSION>.jar
   This product bundles 'os-lib' from the above files.
@@ -1782,7 +1808,7 @@ is subject to the terms and conditions of the following licenses.
   This product bundles 'SLF4J API' from the above files.
   These files are available under the MIT license:
 
-    Copyright (c) 2004-2017 QOS.ch
+    Copyright (c) 2004-2022 QOS.ch Sarl (Switzerland)
     All rights reserved.
 
     Permission is hereby granted, free  of charge, to any person obtaining
```

Reply:
> Oh there's a build/package directory with its own LICENSE and NOTICE file... hmm.

Yeah, the root `LICENSE/NOTICE` files are only for files that are actually in the repo. The `build/package/LICENSE,NOTICE` files are for files that end up in the vsix (e.g. dependency jars).